### PR TITLE
Remove `github.com/gravitational/ttlmap` dependency

### PR DESCRIPTION
**Pull request by Joerger**
Replace `github.com/gravitational/ttlmap` with `utils.FnCache`.

### CODE REVIEW

File: lib/utils/fncache.go [hunk 1]
```diff
@@ -170,6 +170,24 @@ func (c *FnCache) Shutdown(ctx context.Context) {
...
+// Set set an item in the cache.
+func (c *FnCache) Set(key any, v any, ttl time.Duration) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.entries[key] = &fnCacheEntry{
+		v:   v,
+		t:   c.cfg.Clock.Now(),
+		ttl: ttl,
+	}
+}
```
Comment (fspmarshall) [inline lib/utils/fncache.go:189]: We definitely want test coverage for the new methods.  Also, I believe that the entry being added here needs to have its `loaded` channel be a valid closed channel, not nil, in order for subsequent reads & ttl cleanups to work correctly.

File: lib/kube/proxy/forwarder_test.go [hunk 2]
```diff
@@ -1106,7 +1104,10 @@ func TestKubeFwdHTTPProxyEnv(t *testing.T) {
 
 func newMockForwader(ctx context.Context, t *testing.T) *Forwarder {
 	clock := clockwork.NewFakeClock()
-	cachedTransport, err := ttlmap.New(defaults.ClientCacheSize, ttlmap.Clock(clock))
+	cachedTransport, err := utils.NewFnCache(utils.FnCacheConfig{
+		TTL:   5 * time.Hour,
```
Comment (fspmarshall) [inline lib/kube/proxy/forwarder_test.go:1108]: nit: this should be a named constant, or at least have an inline comment explaining what it is (this applies to a few other places as well).

File: lib/bpf/bpf.go [hunk 3]
```diff
@@ -364,27 +363,31 @@ func (s *Service) emitCommandEvent(eventBytes []byte) {
...
-			buf = make([]string, 0)
-		} else {
-			buf = buffer.([]string)
-		}
+		key := strconv.FormatUint(event.PID, 10)
+
+		// loadFn does not return an error, so we can skip error handling.
+		args, _ := utils.FnCacheGet(s.closeContext, s.argsCache, key, func(ctx context.Context) ([]string, error) {
+			return make([]string, 0), nil
+		})
```
Comment (fspmarshall) [inline lib/bpf/bpf.go:371]: nit: it's very easy to accidentally break this invariant elsewhere. Consider at least having a `this is a bug` log get triggered in the event of a non-nil error.

**Review by fspmarshall — APPROVED** [review]
LGTM once the new ttl cache methods have decent test coverage.

**Review by EdwardDowling — APPROVED** [review]

**Review by fspmarshall — APPROVED** [review]

**Comment by Joerger** [comment]
@fspmarshall We had to enabled `ReloadOnErr` for the app session cache due to potentially transient errors creating a new session when no app servers were matched. It doesn't seem like you intended `ReloadOnErr` to be used this way, but IMO it does exactly what we need to handle transient errors.

Alternatively we could validate the session each time before checking the session cache in order to avoid recording failed entries - d82d439b175ea24892578d11ff41717644c55225

WDYT?

**Comment by fspmarshall** [comment]
@Joerger avoiding retention of a transient error sounds like a good use of it to me.  I'm all for it.

**Pull request merged**